common/scala/src/main/scala/whisk/core/entity/EntityPath.scala

```diff
@@ -162,10 +188,20 @@ protected[core] object EntityName {
  * - EntityName: the name of the entity
  * - Version: the semantic version of the resource
  */
-protected[core] case class FullyQualifiedEntityName(path: EntityPath, name: EntityName, version: SemVer) {
-    override def toString = path.addpath(name) + "@" + version.toString
+protected[core] case class FullyQualifiedEntityName(path: EntityPath, name: EntityName, version: Option[SemVer] = None) {
+    override def toString = path.addpath(name) + version.map("@" + _.toString).getOrElse("")
```

Comment:
I think this is potentially wrong - because it will generate a doc id with `@...` suffix which will not be a valid doc id (it will not match anything -- unless there's some later change that now stores the version as part of the doc id).